libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -320,6 +320,58 @@ public new BotFrameworkAdapter Use(IMiddleware middleware)
             await connectorClient.Conversations.DeleteActivityAsync(reference.Conversation.Id, reference.ActivityId);
         }
 
+        /// <summary>
+        /// Attempts to retrieve the token for a user that's in a login flow.
+        /// </summary>
+        /// <param name="context">Context for the current turn of conversation with the user.</param>
+        /// <param name="connectionName">Name of the auth connection to use.</param>
+        /// <param name="magicCode">(Optional) Optional user entered code to validate.</param>
+        /// <returns></returns>
+        public async Task<TokenResponse> GetUserToken(ITurnContext context, string connectionName, string magicCode)
+        {
+            if (context == null)
+                throw new ArgumentNullException(nameof(context));
+            if (string.IsNullOrEmpty(connectionName))
+                throw new ArgumentNullException(nameof(connectionName));
+
+            var client = this.CreateOAuthApiClient(context.Services.Get<IConnectorClient>() as ConnectorClient);
+            return await client.GetUserTokenAsync(context.Activity.From.Id, connectionName, magicCode);
+        }
+
+        /// <summary>
+        /// 
```

Comment:
While this comment is quite concise, I suspect a bit of verbosity here may be a better choice. :)